Automattic/pocket-casts-android
# Extract ratings logic to ratings vm and add tracks

## Description
This extracts rating logic to its own vm and adds a track event.

## Testing Instructions
Make sure everything still works from the testing steps in https://github.com/Automattic/pocket-casts-android/pull/868 with a change that ratings load when the podcast is opened instead of when podcast details are expanded to make it consistent with iOS.

Test Tracks
1. With the rating feature flag enabled, open a podcast
2. Tap on the rating stars view
3. Notice that you see the event: `Tracked: rating_stars_tapped, Properties: {"uuid":"...` where uuid is the uuid of the selected podcast.

## Merge Instructions

Since the feature is guarded with a feature flag, it is safe to merge it into `main`:

- Review this PR
- Change base branch to main 
- Merge the PR.
